resources/scripts/components/dashboard/DashboardContainer.tsx

```diff
@@ -51,36 +64,187 @@ export default () => {
         if (!error) clearFlashes('dashboard');
     }, [error]);
 
+    useEffect(() => {
+        const savedGroups = localStorage.getItem(`groups:${uuid}`);
+        if (savedGroups) {
+            setGroups(JSON.parse(savedGroups));
+        }
+    }, [uuid]);
```

Comment:
This will cause unnecessary re-renders. https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

Reply:
I am not really good at Typescript. 

Did you mean [this](https://github.com/pelican-dev/panel/pull/276/commits/d946b207eb214d3ab484e99446bb81689eadc270)? If not could you make a suggestion.